packages/core/src/components/DropDownMenu/DropDownMenu.tsx

```diff
@@ -0,0 +1,188 @@
+import clsx from "clsx";
+import { useControlled } from "hooks";
+import { HvBaseProps } from "../../types";
+import {
+  StyledRoot,
+  StyledBaseDropDown,
+  StyledButton,
+  StyledPanel,
+} from "./DropDownMenu.styles";
+import { dropDownMenuClasses, HvDropDownMenuClasses } from ".";
+import { isKeypress, keyboardCodes, setId } from "utils";
+import getPrevNextFocus from "utils/focusableElementFinder";
+import { MoreOptionsVertical } from "@hitachivantara/uikit-icons";
+import { HvButtonVariant, HvList, HvListValue } from "components";
+import { useMemo } from "react";
+import { theme } from "@hitachivantara/uikit-styles";
+
+export type HvDropDownMenuItem = {
+  label: React.ReactNode;
+  selected?: boolean;
+  disabled?: boolean;
+  icon?: Function | React.ReactNode;
+  showNavIcon?: boolean;
+};
+
+export type HvDropDownMenuProps = HvBaseProps<HTMLDivElement, { onClick }> & {
+  /** Icon. */
+  icon?: React.ReactElement;
+  /**
+   * A list containing the elements to be rendered.
+   *
+   * - label: The label of the element to be rendered.
+   * - selected: The selection state of the element.
+   * - disabled: The disabled state of the element.
+   * - icon: The icon node to be rendered on the left.
+   * - showNavIcon: If true renders the navigation icon on the right.
+   */
+  dataList: HvDropDownMenuItem[];
+  /** Placement of the dropdown. */
+  placement?: "left" | "right";
+  /** Disable the portal behavior. The children stay within it's parent DOM hierarchy. */
+  disablePortal?: boolean;
+  /** Function executed on toggle of the dropdown. Should receive the open status. */
+  onToggle?: (event: Event, open: boolean) => void;
+  /** Function executed in each onClick. Should received the clicked element. */
+  onClick?: (
+    event: React.ChangeEvent<HTMLLIElement>,
+    value: HvListValue
+  ) => void;
+  /** Keep the Dropdown Menu opened after clicking one option */
+  keepOpened?: boolean;
+  /** Defines if the component is disabled. */
+  disabled?: boolean;
+  /** If true it should be displayed open. */
+  expanded?: boolean;
+  /** When uncontrolled, defines the initial expanded state. */
+  defaultExpanded?: boolean;
+  /** The variant to be used in the header. */
+  category?: HvButtonVariant;
+  /** A Jss Object used to override or extend the styles applied to the empty state component. */
+  classes?: HvDropDownMenuClasses;
+};
+
+/**
+ * A drop-down menu is a graphical control element, similar to a list box, that allows the user to choose a value from a list.
+ */
+export const HvDropDownMenu = ({
```

Comment:
We need to wrap the component with `withId` since the [master branch](https://github.com/lumada-design/hv-uikit-react/blob/master/packages/core/src/DropDownMenu/DropDownMenu.js#L224) does it too. 
I also had to use this wrapper for the `FileUploader` component. Otherwise the accessibility was breaking since the component did not have an id when it was not provided.